src/createServer.js

```diff
@@ -1,8 +1,108 @@
 'use strict';
 
+const http = require('http');
+const path = require('path');
+const fs = require('fs');
+const mime = require('mime-types');
+const zlib = require('zlib');
+const { formidable } = require('formidable');
+const { pipeline } = require('stream');
+
 function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
+  const server = new http.Server();
+
+  server.on('request', (req, res) => {
+    const url = new URL(req.url || '', `http://${req.headers.host}`);
+    const requestedPath = url.pathname.slice(1) || 'index.html';
+
+    if (requestedPath === 'compress' && req.method === 'GET') {
+      res.writeHead(400, { 'content-type': 'text/plain' });
+      res.end('Request should be POST method!');
```

Comment:
When the request method is GET for the 'compress' route, it would be more appropriate to return a 405 Method Not Allowed status code instead of 400 Bad Request. This status code better indicates that the method is not allowed for the requested resource.